Sources/LambdaConverters.Wpf/Validator.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.Windows.Controls;
+using JetBrains.Annotations;
+
+namespace LambdaConverters
+{
+    /// <summary>
+    /// A factory class used to create lambda-based instances of the <see cref="ValidationRule"/> class.
+    /// </summary>
+    public static class Validator
+    {
+        sealed class Rule<I> : System.Windows.Controls.ValidationRule
+        {
+            readonly Func<ValidationRuleArgs<I>, ValidationResult> ruleFunction;
+
+            internal Rule(
+                Func<ValidationRuleArgs<I>, ValidationResult> ruleFunction,
+                RuleErrorStrategy errorStrategy)
+            {
+                ErrorStrategy = errorStrategy;
+                IsRuleFunctionAvailable = ruleFunction != null;
+                this.ruleFunction = ruleFunction;
+            }
+
+            RuleErrorStrategy ErrorStrategy { get; }
+
+            bool IsRuleFunctionAvailable { get; }
```

Comment:
The private property `IsRuleFunctionAvailable` looks also redundant. Consider removing the property.